tools/firtool/firtool.cpp

```diff
@@ -342,6 +345,14 @@ static cl::opt<std::string>
                 cl::init(""), cl::value_desc("filename"),
                 cl::cat(mainCategory));
 
+static cl::opt<bool>
+    emitBytecode("emit-bytecode",
+                 cl::desc("Emit bytecode when generating MLIR output"),
+                 cl::init(false), cl::cat(mainCategory));
+
+static cl::opt<bool> force("f", cl::desc("Enable binary output on terminals"),
+                           cl::init(false), cl::cat(mainCategory));
```

Comment:
What is the use case of force flag?

Reply:
Overriding overly clever or "I know better" utilities in situations not anticipated :sweat_smile: .
Borrowed the notion from LLVM utilities, which goes back to https://llvm.org/PR305 (and indeed early testing I wrote it to my terminal).

It's a good question, thank you.
I think the `-f` flag (and tty check) makes more sense if it can happen by default/accident, like with `opt`, but since we are opt-in for now this might be unnecessary.  FWIW `mlir-opt` (and `circt-opt`) will presently happily print bytecode to stdout when requested with `-emit-bytecode`.  `circt-opt` uses MLIR's "producer" string but I don't think this is a big deal.  The `-f` flag is mentioned by the error message presently.

The tty check from LLVM is more LLVM-specific than I realized at first ("LLVM bitcode") so I'll rework this anyway.

> Awesome, thank you for quickly adapting bitcode

Thanks and thanks for quick review! Very welcome, fun feature to incorporate and likely useful in a number of ways to us.

Micro-nit, and mostly because I got it wrong myself a few times, it's bytecode for MLIR's binary format, not bitcode (LLVM's).